libs/telemetry/src/main/java/org/opensearch/telemetry/metrics/ObservableMeasurement.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.metrics;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.telemetry.metrics.tags.Tags;
+
+/**
+ * Observable Measurement for the Asynchronous instruments.
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class ObservableMeasurement {
```

Comment:
I don't think we have to copy the OTEL naming convention, `Metric` or `TaggedMetric` would probably be fine, wdyt?

Reply:
`TaggedMeasurement` is more appropriate.